plugin/core/windows.py

```diff
@@ -557,6 +566,43 @@ def _check_window_closed(self) -> None:
         if not self._is_closing and not self._window.is_valid():
             self._handle_window_closed()
 
+    def _receive_progress_token(self, params: Dict[str, Any], client: Client, request_id: Any) -> None:
+        self._progress[params['token']] = dict()
+        client.send_response(Response(request_id, None))
+
+    def _handle_progress_notification(self, params: Dict[str, Any]) -> None:
+        token = params['token']
+        if token not in self._progress:
+            debug('unknown $/progress token: {}'.format(token))
+            return
+        value = params['value']
+        if not value:
+            return
+        if value['kind'] == 'begin':
+            self._progress[token]['title'] = value['title']  # mandatory
+            self._progress[token]['message'] = value.get('message')  # optional
+            self._sublime.status_message(self._progress_string(token, value))
```

Comment:
I'm chiming in late on that, but that status message should probably be bound to the view.
window-bound statuses suck as they override each other if there is more than one.

Those notifications can be used both for file-bound (getting diagnostics) or project-bound (indexing) operations but still I would think view-bound status would be better.

Reply:
> window-bound statuses suck as they override each other if there is more than one.

That's a limitation of ST. I think we do the right thing here.

Reply:
Yes, but we are catering for ST, not the general population of editors, so we can choose whatever works best rather than ideal solution. :)

Reply:
That's true, but server-initiated progress is explicitly per-window as indicated by the `window/` prefix of the `window/workDoneProgress/create` request. If the user closes the view while the server is indexing a massive project for 10 minutes then he/she won't see any more progress.